### PR TITLE
(CDAP-16280) Remove methods that promote inefficient usages

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by yaojiefeng — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java [hunk 1]
```diff
@@ -485,4 +471,25 @@ WorkflowStatistics getWorkflowStatistics(WorkflowId workflowId, long startTime,
...
+   * Ensures the given program exists in the given application spec.
+   *
+   * @throws NotFoundException if the program does not exists.
+   */
+  static void ensureProgramExists(ProgramId programId,
+                                  @Nullable ApplicationSpecification appSpec) throws NotFoundException {
+    if (appSpec == null) {
+      throw new ApplicationNotFoundException(programId.getParent());
+    }
+    if (!Objects.equals(programId.getApplication(), appSpec.getName())
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java:485]: check namespace too?

**Review by albertshau — APPROVED** [review]
one minor comment

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java:485]: ApplicationSpecification doesn't carry Namespace information.

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT7107-1

**Pull request merged**